include/boost/geometry/iterators/point_iterator.hpp

```diff
@@ -245,33 +246,26 @@ struct points_end<MultiPolygon, multi_polygon_tag>
 // MK:: need to add doc here
 template <typename Geometry>
 class point_iterator
-    : public detail::point_iterator::iterator_type<Geometry>::type
+    : public boost::iterator_adaptor <
+        point_iterator<Geometry>
+      , typename detail::point_iterator::iterator_type<Geometry>::type
+    >
 {
 private:
     typedef typename detail::point_iterator::iterator_type<Geometry>::type base;
 
-    inline base* base_ptr()
-    {
-        return this;
-    }
-
-    inline base const* base_ptr() const
-    {
-        return this;
-    }
-
     template <typename OtherGeometry> friend class point_iterator;
     template <typename G> friend inline point_iterator<G> points_begin(G&);
     template <typename G> friend inline point_iterator<G> points_end(G&);
 
-    inline point_iterator(base const& base_it) : base(base_it) {}
+    inline point_iterator(base const& base_it) : point_iterator::iterator_adaptor_(base_it) {}
```

Comment:
This line seems too long (more than 80 characters). Please re-write it as:
```
inline point_iterator(typename point_iterator::base_type const& base_it)
    : point_iterator::iterator_adaptor_(base_it)
{}
```
Notice that I also used `base_type` from `boost::iterator_adaptor<>` that I have mentioned in another line comment.